internal/auth0/clients/data_source.go

```diff
@@ -0,0 +1,161 @@
+package clients
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"strings"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/auth0/terraform-provider-auth0/internal/auth0/client"
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+)
+
+// NewDataSource will return a new auth0_clients data source.
+func NewDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: readClientsForDataSource,
+		Description: "Data source to retrieve a list of Auth0 application clients with optional filtering.",
+		Schema: map[string]*schema.Schema{
+			"name_filter": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Filter clients by name (partial matches supported).",
+			},
+			"app_types": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "Filter clients by application types.",
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringInSlice(client.ValidAppTypes, false),
+				},
+			},
+			"is_first_party": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Filter clients by first party status.",
+			},
+			"clients": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "List of clients matching the filter criteria.",
+				Elem: &schema.Resource{
+					Schema: CoreClientDataSourceSchema(),
+				},
+			},
+		},
+	}
+}
+
+func CoreClientDataSourceSchema() map[string]*schema.Schema {
```

Comment:
I don't think this needs to be public, so changing it to coreClientDataSourceSchema will clear up the lint error

Reply:
Good catch. Updated as suggested.